hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruCachedBlockQueue.java

```diff
@@ -65,14 +65,15 @@ public LruCachedBlockQueue(long maxSize, long blockSize) {
    * @param cb block to try to add to the queue
    */
   public void add(LruCachedBlock cb) {
-    if(heapSize < maxSize) {
+    long cbSize = cb.heapSize();
+    if(heapSize + cbSize < maxSize) {
       queue.add(cb);
-      heapSize += cb.heapSize();
+      heapSize += cbSize;
     } else {
       LruCachedBlock head = queue.peek();
       if(cb.compareTo(head) > 0) {
         heapSize += cb.heapSize();
-        heapSize -= head.heapSize();
+        heapSize -= cbSize;
```

Comment:
why does this line change? IIUC, we are removing the current 'head' and adding the input cache block, so the existing code makes sense?

or did you mean to change the line above ?

Reply:
Oh yes, my bad!